.bazelci/presubmit.yml

```diff
@@ -610,6 +610,21 @@ tasks:
   #   working_directory: examples/zig_cross_compiling
   #   build_targets:
   #     - "//..."
+  nix_cross_compiling:
+    name: Nix cross compiling test
+    platform: ubuntu2204
+    working_directory: examples/nix_cross_compiling
+    setup:
```

Comment:
This is something that can be flushed into the container now that you know what works, right?

Reply:
I'm not sure what you mean by "flushed into the container"?  This is just using one of the normal Ubuntu containers now and I've updated my latest PR in the other repo to remove the "Ubuntu (Nix)" image: https://github.com/bazelbuild/continuous-integration/pull/1827

Reply:
If you mean baked into the image, no, not reasonably.  Having Nix ship with the image only saves a matter of seconds.  Almost all of the time is spent building `cargo-bazel` and fetching the toolchains.  Baking those into the image would be difficult and brittle, as either their definition needs to live in that repo and somehow be referenced here, or as soon as the respective lockfiles are out of sync, there will no longer be a cache hit on the Nix store in the image and we're back to where we are now.

Reply:
I think I'm confused, how is this setup doing anything other than machine configuration? Looks like you're installing nix, copying some stuff around and updating some config file? This to me feels like something that could go into the new nix image. But doesn't need to be a PR blocker. The commands looked like it took 13 seconds so no issue there.

Reply:
The problem there is that Nix has two modes, single user and multi-user.  Multi-user means install as `root` and run `nix daemon` to handle requests from all users.  This is turn means having an entrypoint script in the container to start the daemon and extra complexity over single user mode.  For single user, Nix needs to be installed by that user, which meant the `Dockerfile` either needs to have ahead of time knowledge that BuildKite will use 997:997 (UID/GID) or that the installation needed to happen as part of the test run.  I've opted for the latter as that seems like the simplest and most robust option.